src/main/ruby/core/posix.rb

```diff
@@ -217,6 +219,41 @@ def self.unsetenv(name)
   else
     raise 'Unsupported platform'
   end
+  
+  def self.write_string(fd, string, nonblock=false)
+    # TODO CS 14-Nov-17 this involves copying to a Java byte[], and then NFI will copy it again!
+    bytes = Truffle.invoke_primitive :string_to_null_terminated_byte_array, string
```

Comment:
Just a note: in this case we don't need it to be null-terminated.

Reply:
Using `:'[sint8]'` will also copy the bytes on the way back on JVM, so it might be more efficient to actually use a buffer (possibly thread-local) and pass it as a Pointer, like you did before.